src/pymanifold.py

```diff
@@ -125,6 +125,54 @@ def channel(self,
             self.dg.edges[port_from, port_to][key] = attr
         return
 
+    def retrieve(port_in, port_out, attr):
```

Comment:
This can simplified to be ```def retrieve(port_in, attr, port_out=None):``` so that each of the following calls to this method wouldn't need to specify that port_out equals None if they're only providing one node.